### PR TITLE
Added SDL2 audio

### PR DESCRIPTION
**Pull request by mir3x**

### CODE REVIEW

File: cmake/FindSDL2.cmake [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  endif()
+
+  # For OS X, SDL uses Cocoa as a backend so it must link to Cocoa.
+  # CMake doesn't display the -framework Cocoa string in the UI even
+  # though it actually is there if I modify a pre-used variable.
+  # I think it has something to do with the CACHE STRING.
+  # So I use a temporary variable until the end so I can set the
+  # "real" variable in one-shot.
+  if(APPLE)
+    set(SDL_LIBRARY_TEMP ${SDL2_LIBRARY_TEMP} "-framework Cocoa")
```
Comment (lmoureaux) [inline cmake/FindSDL2.cmake:140]: This should be `SDL_LIBRARY_TEMP`, or all `SDL_LIBRARY_TEMP` should become `SDL2_LIBRARY_TEMP`.

File: cmake/FreecivDependencies.cmake [hunk 2]
```diff
@@ -51,6 +51,20 @@ if(FREECIV_ENABLE_NLS)
   set(FREECIV_HAVE_LIBINTL_H TRUE)
 endif()
 
+# SDL2 for audio
+find_package(SDL2 QUIET)
+find_package(SDL2_mixer QUIET)
+if (SDL2_MIXER_LIBRARIES AND SDL2_LIBRARY)
+  set(AUDIO_SDL TRUE)
+  set(SDL2_PLAIN_INCLUDE TRUE)
```
Comment (lmoureaux) [inline cmake/FreecivDependencies.cmake:59]: Better drop the `#ifdef` in `client/audio_sdl.cpp`

File: utility/cmake_fc_config.h.in [hunk 3]
```diff
@@ -41,9 +41,7 @@
 /* enable audio */
 #cmakedefine AUDIO_SDL
 
-/* audio via SDL2_mixer */
-#cmakedefine AUDIO_SDL2
-
+#cmakedefine SDL2_PLAIN_INCLUDE
```
Comment (lmoureaux) [inline utility/cmake_fc_config.h.in:44]: Drop the `#ifdef` instead

File: cmake/FreecivDependencies.cmake (hunk 2 above)
Comment (mir3x) [reply cmake/FreecivDependencies.cmake]: updated

Comment (lmoureaux) [reply cmake/FreecivDependencies.cmake]: Waiting for Travis to start...

**Pull request merged**